test/utils/Checkpoints.t.sol

```diff
@@ -0,0 +1,276 @@
+// SPDX-License-Identifier: MIT
+// This file was procedurally generated from scripts/generate/templates/Checkpoints.test.js.
+
+pragma solidity ^0.8.0;
+
+import "forge-std/Test.sol";
+import "../../contracts/utils/Checkpoints.sol";
+import "../../contracts/utils/math/SafeCast.sol";
+
+contract CheckpointsHistoryTest is Test {
+    using Checkpoints for Checkpoints.History;
+
+    Checkpoints.History internal _ckpts;
+
+    // helpers
+    function _boundUint32(uint32 x, uint32 min, uint32 max) internal view returns (uint32) {
+        return SafeCast.toUint32(bound(uint256(x), uint256(min), uint256(max)));
+    }
+
+    function _prepareKeys(uint32[] memory keys, uint32 maxSpread) internal view {
+        uint32 lastKey = 0;
+        for (uint256 i = 0; i < keys.length; ++i) {
+            uint32 key = _boundUint32(keys[i], lastKey, lastKey + maxSpread);
+            keys[i] = key;
+            lastKey = key;
+        }
+    }
+
+    function _assertLatestCheckpoint(bool exist, uint32 key, uint224 value) internal {
+        (bool _exist, uint32 _key, uint224 _value) = _ckpts.latestCheckpoint();
+        assertTrue(_exist == exist);
+        assertTrue(_key == key);
+        assertTrue(_value == value);
+    }
+
+    // tests
+    function testPush(uint32[] memory keys, uint224[] memory values) public {
+        vm.assume(values.length > 0);
```

Comment:
Do we want to add this? Feels weird that without this condition Foundry will invest time into generating cases that we will not fully consider, and will count them as successful fuzz runs, and we may have wanted to use those runs to explore more of the "useful" argument space.

This applies to `testLookup` as well.

```suggestion
        vm.assume(values.length > 0 && values.length <= keys.length);
```